src/core/localize.rkt

```diff
@@ -22,6 +23,23 @@
                (for ([arg args] [atype atypes])
                  (loop arg atype))])))))
 
+(define (all-subexpressions-rev expr repr)
```

Comment:
Why not reverse of all-subexpressions?

Reply:
The issue with the `all-subexpressions` was, it removes duplicates at the wrong place. For example, for the expression √x - √x + 1, I'd get the list [√x - √x + 1, √x, x, √x + 1, x + 1, 1], and now when I reverse it I'd get [1, x + 1, √x + 1, x, √x, √x - √x + 1]; I lose the information that x is child of x + 1. My original thought was that if I want to find the child expression that under/overflowed, I'd have to delete duplicates in reverse so that the child information is kept intact. Regardless I don't think I need to do this anymore, so I can remove it.

Reply:
Ok, I think we could do that as a post-pass so order isn't relevant.